code/_onclick/hud/alert.dm

```diff
@@ -507,6 +514,35 @@ so as to remain in compliance with the most up-to-date laws."
 	timeout = 300
 	var/atom/target = null
 	var/action = NOTIFY_JUMP
+	var/show_time_left = FALSE // If true you need to call START_PROCESSING manually
+	var/image/time_left_overlay // The last image showing the time left
+	var/datum/candidate_poll/poll // If set, on Click() it'll register the player as a candidate
+
+/obj/screen/alert/notify_action/process()
+	if(show_time_left)
+		var/timeleft = timeout - world.time
+		if(timeleft <= 0)
+			return PROCESS_KILL
+
+		if(time_left_overlay)
+			overlays -= time_left_overlay
+
+		var/obj/O = new
+		O.maptext = "<span style='font-weight: bold; font-size: 32px; color: [(timeleft <= 10 SECONDS) ? "red" : "white"];'>[CEILING(timeleft / 10, 1)]</span>"
```

Comment:
Here

Reply:
Done and boy is it c r i s p

![image](https://user-images.githubusercontent.com/1496804/89959679-4c555d00-dc3d-11ea-943b-1d6ac82a4702.png)